ReactiveCocoa/Swift/Signal.swift

```diff
@@ -139,6 +143,12 @@ public final class Signal<Value, Error: ErrorType> {
 			return nil
 		}
 	}
+	
+	deinit {
+		if atomicObservers.value?.count ?? 0 == 0 {
```

Comment:
Having observers means the event stream isn't really ended. `Signal` here is just an interface to attach observers. Say:
```
func foo() {
    someExternalSignal
        .map { String("value: \($0)") }
        .observeNext { print($0) }
}

foo()
```
Upon exiting `foo()`, the "shell" of the mapped signal would have been deallocated. But the event stream should still go on until a terminating event hits, or all the observers are detached from the mapped signal, i.e. an empty bag of observers.